doc/1.4/language.md

```diff
@@ -1680,8 +1688,16 @@ In the absence of explicit initializer expression, a default
 Note that the number of elements in the initializer must match with
 the number of elements or fields of the type of the *saved*
 variable. This also implies that each sub-element, if itself being a
-compound data structure, must also be enclosed in braces. C99-style
-designated initializers are not supported.
+compound data structure, must also be enclosed in braces.
+
+C99-style designated initializers are supported for `struct`, `layout`, and
+`bitfields` types:
+```
+typedef struct { int x; struct { int i; int j; } y; } struct_t;
+saved struct_t s = { .x = 1, .y = { .i = 2, .j = 3 } }
```

Comment:
Documentation should also mention that initializers are available in assignment statements (not strictly needed for this patch, but related since you duplicate information).

Reply:
In general, we need a section in the manual for assign statements, where we can state that initializers can be used as RHS of assigns; move stuff from the "Comparison to C/C++" section to that section; and move the explanations of the various forms of initializers throughout the manual to that section. This isn't appropriate to do in this PR, however. I'll create an issue about it.